liblll/CodeFragment.cpp

```diff
@@ -192,7 +192,13 @@ void CodeFragment::constructOperation(sp::utree const& _t, CompilerState& _s)
 		{
 			if (_t.size() != 2)
 				error<IncorrectParameterCount>();
-			m_asm.append(CodeFragment::compile(contentsString(firstAsString()), _s).m_asm);
+			string fileName = firstAsString();
+			if (fileName.empty())
+				error<InvalidName>("Empty file name provided");
+			string contents = contentsString(fileName);
```

Comment:
I did not know this was here. This permits reading any file from the filesystem. If lll is added to the json compiler, this security flaw is extended to solidity.